MarkupsToModel/Logic/vtkSlicerMarkupsToModelLogic.cxx

```diff
@@ -538,6 +577,373 @@ void vtkSlicerMarkupsToModelLogic::UpdateOutputHermiteSplineModel(vtkMRMLMarkups
 }
 
 //------------------------------------------------------------------------------
+void vtkSlicerMarkupsToModelLogic::ComputePointParametersRawIndices(vtkPolyData * markupsPointsPolyData, vtkDoubleArray* markupsPointsParameters)
+{
+  vtkPoints* points = markupsPointsPolyData->GetPoints();
+  if (points == NULL)
+  {
+    vtkErrorMacro("Markups contains a vtkPoints object that is null.");
+    return;
+  }
+
+  int numPoints = points->GetNumberOfPoints();
+  // redundant error checking, to be safe
+  if (numPoints < LINE_MIN_NUMBER_POINTS)
+  {
+    vtkErrorMacro("Not enough points to compute polynomial parameters. Need at least " << LINE_MIN_NUMBER_POINTS << " points but " << numPoints << " are provided.");
+    return;
+  }
+
+  if (markupsPointsParameters->GetNumberOfTuples())
+  {
+    // this should never happen, but in case it does, output a warning
+    vtkWarningMacro("markupsPointsParameters already has contents. Clearing.");
+    while (markupsPointsParameters->GetNumberOfTuples()) // clear contents just in case
+      markupsPointsParameters->RemoveLastTuple();
+  }
+
+  for (int v = 0; v < numPoints; v++)
+  {
+    markupsPointsParameters->InsertNextTuple1( v / double(numPoints-1) );
+    // division to clamp all values to range 0.0 - 1.0
+  }
+}
+
+//------------------------------------------------------------------------------
+void vtkSlicerMarkupsToModelLogic::ComputePointParametersMinimumSpanningTree(vtkPolyData * markupsPointsPolyData, vtkDoubleArray* markupsPointsParameters)
+{
+  vtkPoints* points = markupsPointsPolyData->GetPoints();
+  if (points == NULL)
+  {
+    vtkErrorMacro("Markups contains a vtkPoints object that is null.");
+    return;
+  }
+
+  int numPoints = points->GetNumberOfPoints();
+  // redundant error checking, to be safe
+  if (numPoints < LINE_MIN_NUMBER_POINTS)
+  {
+    vtkErrorMacro("Not enough points to compute polynomial parameters. Need at least " << LINE_MIN_NUMBER_POINTS << " points but " << numPoints << " are provided.");
+    return;
+  }
+
+  // vtk boost algorithms cannot be used because they are not built with 3D Slicer
```

Comment:
when you need a long comment like this, it indicates that the method should be split into 5 distinct methods - of course that many additional methods in this already busy class would not be nice => one more reason to move all polyfit related methods into a separate class